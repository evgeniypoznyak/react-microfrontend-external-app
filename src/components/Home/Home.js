import React from 'react';

const Home = () => {
  return <div>Hi! I'm a external component from another server!</div>;
};
export default Home;
