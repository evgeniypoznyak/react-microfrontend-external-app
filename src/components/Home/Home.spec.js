import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from './Home';
import { mount, shallow } from 'enzyme';

let wrapper;
let mountWrapper;

beforeAll(() => {
  wrapper = shallow(<Home />);
  mountWrapper = mount(<Home />);
});

describe('<Home/>', () => {
  it('renders correctly', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
    expect(wrapper.children.length).toBe(1);
  });

  it('Should contain text "external" and "Hi" ', () => {
    const { container } = render(<Home />);
    expect(screen.getByText(/external/i).textContent).toContain('Hi');
    expect(container.firstChild.textContent).toContain('Hi');
  });

  it('should render Home with children', () => {
    const app = mountWrapper.find(Home);
    expect(app).toHaveLength(1);
    expect(app.children).toHaveLength(1);
  });
});
