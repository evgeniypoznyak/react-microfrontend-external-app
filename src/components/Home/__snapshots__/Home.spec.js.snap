// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> renders correctly 1`] = `
<div>
  <div>
    Hi! I'm a external component from another server!
  </div>
</div>
`;
